kv/union_store.go

```diff
@@ -39,6 +39,23 @@ type UnionStore interface {
 	GetMemBuffer() MemBuffer
 }
 
+// ContractType is the type of a contract.
+type ContractType int
+
+// The ContractType constants.
+const (
```

Comment:
This is called invariance properties In TLA+, i prefer to use the same name, Contract makes me confused. Please see the original paper: https://lamport.azurewebsites.net/tla/charme.ps.Z

Reply:
Invariance seems not frequently used.
How about constraint?

Reply:
`Assumption` seems better to describe the logic.
When we put or delete a key, we assume the key exist or not exist.

Reply:
How about `assertion`? I think `constraint` is ok, but SQL has the concept of constraint too. `Assumption` is not often used. `invariance properties` obviously not suitable.

Reply:
I changed the name to `Assumption` just now